File: Medium/Maximum_Subarray/Maximum_Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSubSum = nums[0];
        int sum = 0;
        for (int num : nums) {
            sum += num;
            if (sum < num) {
                maxSubSum = Math.max(maxSubSum, num);
                sum = num;
            }
            maxSubSum = Math.max(maxSubSum, sum);
        }

        return Math.max(maxSubSum, sum);
    }
}

File: Medium/Maximum_Subarray/Maximum_Subarray.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxSubArray(nums) {
        let maxSubSum = nums[0];
        let currSum = 0;

        for (let num of nums) {
            currSum += num;
            if (currSum < num) {
                currSum = num;
            }
            maxSubSum = Math.max(maxSubSum, currSum);
        }

        return maxSubSum;
    }
}


